.github/workflows/reusable-e2e.yaml

```diff
@@ -0,0 +1,191 @@
+name: Reusable e2e workflow
+
+on:
+  workflow_call:
+    inputs:
+      operator:
+        required: true
+        type: boolean
+
+jobs:
+  operator_e2e:    
+    if: ${{inputs.operator}}
+    runs-on: ubuntu-latest
+    name: e2e
+    env:
+      MIRRORD_AGENT_RUST_LOG: "warn,mirrord=debug"
+      MIRRORD_OPERATOR_IMAGE: "test"
+    steps:            
+      - uses: actions/checkout@v3
+        with:
+          repository: "metalbear-co/mirrord"        
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly-2023-04-19
+          components: rustfmt, clippy
+      - uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: build mirrord
+        run: cargo build --manifest-path=./Cargo.toml
+      - name: download operator image
+        uses: actions/download-artifact@v3
+        with:
+          name: test-operator
+          path: /tmp
+      - name: start minikube
+        uses: medyagh/setup-minikube@master
+        with:
+          container-runtime: containerd
+      - run: minikube image load /tmp/test-operator.tar
+      - run: minikube image list
+      - name: setup operator
+        run: |
+          target/debug/mirrord operator setup --accept-tos --license-path ./tests/test-license.pem | kubectl apply -f -              
```

Comment:
Need to patch the deployment to use a different agent image (by default it will use by crate semver) via MIRRORD_AGENT_IMAGE on the deployment like 

Something like
```bash
kubectl -n mirrord patch deployment mirrord-operator -p '{"spec":{"template":{"spec":{"containers":[{"name":"mirrord-operator","env": [{"name":"MIRRORD_AGENT_IMAGE","value":"test"}]}]}}}}'
```